### PR TITLE
Prioritize endpoint config over configuration file

### LINKED ISSUE #1634: Endpoint config should not be overwritten by Zenoh config file
### Describe the bug

Some link configurations can either be passed by endpoint parameters or via the zenoh config file. Currently the config file is given priority over endpoint parameters, which shouldn't be the case. Config file should be a fallback if no config is given on the specific endpoint.

### To reproduce

1. Create a zenoh config file with TLS config.
2. Set `transport/link/tls/enable_mtls` to true.
3. Set `listen/endpoints` to `["tls/[::]:7447#enable_mtls=false"]`
4. Start a peer/router with the config file, connect to it with a client via tls and notice how mtls is enabled.

### System info

Zenoh 1.0.4

### PR DESCRIPTION
**Pull request by oteffahi**
Closes #1634

### CODE REVIEW

File: commons/zenoh-protocol/src/core/endpoint.rs [hunk 1]
```diff
@@ -385,8 +385,9 @@ impl ConfigMut<'_> {
             self.0.address(),
             self.0.metadata(),
             parameters::from_iter(parameters::sort(parameters::join(
-                self.0.config().iter(),
+                // Endpoint config overwrites incoming config
                 iter.map(|(k, v)| (k.borrow(), v.borrow())),
+                self.0.config().iter(),
```
Comment (Mallets) [inline commons/zenoh-protocol/src/core/endpoint.rs:390]: I don't think this is the right change to solve the issue. 
IMHO this changes is wrong w.r.t. the expected behaviour of this function where the iter in this is overwritten by the config. 

The change should be done in the transport like here: https://github.com/eclipse-zenoh/zenoh/blob/233240622f548c18fe41450144010ad1b7e1fe17/io/zenoh-transport/src/unicast/manager.rs#L386.

Comment (oteffahi) [reply commons/zenoh-protocol/src/core/endpoint.rs:390]: This function is currently only used for merging config file with endpoint config. Would renaming it to indicate its purpose solve the issue then?

File: io/zenoh-transport/zenoh-test-unix-socket-0.sock.lock
Comment (Mallets) [inline io/zenoh-transport/zenoh-test-unix-socket-0.sock.lock:1]: Remove this file

File: io/zenoh-transport/zenoh-test-unix-socket-1.sock.lock
Comment (Mallets) [inline io/zenoh-transport/zenoh-test-unix-socket-1.sock.lock:1]: Remove this file

Comment (oteffahi) [reply io/zenoh-transport/zenoh-test-unix-socket-1.sock.lock:1]: oops 😅

File: io/zenoh-transport/zenoh-test-unix-socket-0.sock.lock
Comment (oteffahi) [reply io/zenoh-transport/zenoh-test-unix-socket-0.sock.lock:1]: oops 😅

**Review by Mallets — APPROVED** [review]

File: zenoh/tests/tcp_buffers.rs
Comment (Mallets) [inline zenoh/tests/tcp_buffers.rs:1]: This test shouldn't be part of `zenoh/tests` but rather in `zenoh-links` itself or `zenoh-transport`.

Comment (Mallets) [reply zenoh/tests/tcp_buffers.rs:1]: All link-specific tests are part of `zenoh-transport`. We can move the test in a different PR.

**Pull request merged**